app/server/src/main/scala/org/bitcoins/server/BitcoinSRunner.scala

```diff
@@ -68,32 +69,30 @@ trait BitcoinSRunner extends BitcoinSLogger {
         .resolve()
   }
 
-  lazy val configDataDir: Path = Paths.get(
-    baseConfig.getStringOrElse("bitcoin-s.datadir",
-                               AppConfig.DEFAULT_BITCOIN_S_DATADIR.toString))
-
-  lazy val datadirPath: Path = datadirPathOpt.getOrElse(configDataDir)
-
-  private lazy val networkStr: String =
-    baseConfig.getString("bitcoin-s.network")
-
-  private lazy val network: BitcoinNetwork =
-    BitcoinNetworks.fromString(networkStr)
-
-  lazy val datadir: Path = {
-    lazy val lastDirname = network match {
-      case MainNet  => "mainnet"
-      case TestNet3 => "testnet3"
-      case RegTest  => "regtest"
-      case SigNet   => "signet"
-    }
-    datadirPath.resolve(lastDirname)
-  }
+  /** Base directory for all bitcoin-s data */
+  lazy val configDataDir: Path =
+    Paths.get(baseConfig.getString("bitcoin-s.datadir"))
 
   def startup: Future[Unit]
 
   // start everything!
   final def run(): Unit = {
+
+    /** Directory specific for current network */
+    val datadir: Path = {
+      val networkStr: String =
+        baseConfig.getString("bitcoin-s.network")
+      val network: BitcoinNetwork =
+        BitcoinNetworks.fromString(networkStr)
+      lazy val lastDirname = network match {
+        case MainNet  => "mainnet"
+        case TestNet3 => "testnet3"
+        case RegTest  => "regtest"
+        case SigNet   => "signet"
+      }
+      configDataDir.resolve(lastDirname)
```

Comment:
Does this still take into account the ``--datadir` command line option? Where does that come into play here?

Reply:
`configDataDir` is the resulting datadir from the `--datadir` `--conf` and `bitcoin-s.conf`